example-app/App.tsx

```diff
@@ -40,14 +40,45 @@ const Header = () => {
 };
 
 const App = () => {
+  const [result, setResult] = React.useState<DeepLinkResult>(undefined);
+
+  React.useEffect(() => {
+    Linking.addEventListener('url', event => {
+      setResult(handleDeepLink(event.url));
+    });
+
+    async function getDeepLink() {
+      let url = await Linking.getInitialURL();
+      let res = handleDeepLink(url);
+      setResult(res);
+    }
+
+    getDeepLink();
+  }, []);
+
+  const handleDeepLink = (url: string | null): DeepLinkResult => {
+    if (!url) {
+      return;
+    }
+
+    let parsedQs = qs.parse(url.substring(url.indexOf('?') + 1));
+    let error = parsedQs.error;
+
+    if (error) {
+      return {success: false, errorReason: error.toString()!};
```

Comment:
What is the type of `error`? Would be good to avoid `.toString()` and especially the non null assertion. This is example code, so it needs to look good. No shortcuts or hacks.

Reply:
I'll revert to previous solution w url-parse